src/reformat.c

```diff
@@ -505,7 +505,7 @@ avifResult avifImageRGBToYUV(avifImage * image, const avifRGBImage * rgb)
                         }
                     }
                 }
-                g = avifRoundf(AVIF_CLAMP(g * state.yuv.rangeY, 0.0f, state.yuv.maxChannel));
+                g = avifYUVColorSpaceInfoYToUNorm(&state.yuv, g);
```

Comment:
Nit: `avifYUVColorSpaceInfoYToUNorm()` returns `int`. I suggest storing the return value in an `int` variable `y` to avoid the unnecessary conversion to `float` and back to an integer.

```
                int y = avifYUVColorSpaceInfoYToUNorm(&state.yuv, g);
```

And then use `y` instead of `g` in line 511 and line 513.

Reply:
done